src/app/features/courses/components/courses-list/components/courses-search/courses-search.component.ts

```diff
@@ -1,21 +1,51 @@
-import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import {
+  AfterViewInit,
+  Component,
+  ElementRef,
+  EventEmitter,
+  OnDestroy,
+  Output,
+  ViewChild,
+} from '@angular/core';
+import { fromEvent, map, filter, debounceTime, Subject, takeUntil } from 'rxjs';
 
 @Component({
   selector: 'app-courses-search',
   templateUrl: './courses-search.component.html',
   styleUrls: ['./courses-search.component.scss'],
 })
-export class CoursesSearchComponent implements OnInit {
-  searchValue: string = '';
+export class CoursesSearchComponent implements AfterViewInit, OnDestroy {
+  private readonly destroy$ = new Subject<boolean>();
 
   @Output()
   searchValueChange = new EventEmitter<string>();
 
+  @ViewChild('searchInput')
+  searchInput!: ElementRef;
```

Comment:
You can declare it this way:
`@ViewChild('searchInput', { static: true }) searchInput: ElementRef<HTMLElement>;`
 Because, you don't have any dynamic data (for example, generated lists by ngFor directive), you can access ViewChild  in `OnInit` hook, if you pass `{ static: true }`.

Read more [here](https://itecnote.com/tecnote/angular-v8-viewchild-static-true-or-false/).